runtime/src/snapshot_utils.rs

```diff
@@ -649,64 +896,280 @@ pub fn bank_from_snapshot_archive<P>(
     )?;
-    measure.stop();
+    measure_rebuild.stop();
+    info!("{}", measure_rebuild);
 
-    let mut verify = Measure::start("verify");
+    let mut measure_verify = Measure::start("verify");
     if !bank.verify_snapshot_bank(test_hash_calculation)
         && limit_load_slot_count_from_snapshot.is_none()
     {
         panic!("Snapshot bank for slot {} failed to verify", bank.slot());
     }
-    verify.stop();
+    measure_verify.stop();
+
     let timings = BankFromArchiveTimings {
-        rebuild_bank_from_snapshots_us: measure.as_us(),
-        untar_us: untar.as_us(),
-        verify_snapshot_bank_us: verify.as_us(),
+        rebuild_bank_from_snapshots_us: measure_rebuild.as_us(),
+        untar_us: full_snapshot_measure_untar.as_us()
+            + incremental_snapshot_measure_untar.map_or(0, |measure| measure.as_us()),
+        verify_snapshot_bank_us: measure_verify.as_us(),
     };
-
     Ok((bank, timings))
 }
 
+/// Perform the common tasks when unarchiving a snapshot.  Handles creating the temporary
+/// directories, untaring, reading the version file, and then returning those fields plus the
+/// unpacked append vec map.
+fn do_snapshot_unarchive_preparation<P, Q>(
+    snapshot_path: P,
+    unpacked_snapshots_dir_prefix: &'static str,
+    snapshot_archive_path: Q,
+    measure_name: &'static str,
+    account_paths: &[PathBuf],
+    archive_format: ArchiveFormat,
+    parallel_divisions: usize,
+) -> Result<(TempDir, PathBuf, UnpackedAppendVecMap, String, Measure)>
+where
+    P: AsRef<Path>,
+    Q: AsRef<Path>,
+{
+    let unpack_dir = tempfile::Builder::new()
+        .prefix(unpacked_snapshots_dir_prefix)
+        .tempdir_in(snapshot_path)?;
+    let unpacked_snapshots_dir = unpack_dir.path().join("snapshots");
+
+    let mut measure_untar = Measure::start(measure_name);
+    let unpacked_append_vec_map = untar_snapshot_in(
+        snapshot_archive_path,
+        unpack_dir.path(),
+        account_paths,
+        archive_format,
+        parallel_divisions,
+    )?;
+    measure_untar.stop();
+    info!("{}", measure_untar);
+
+    let unpacked_version_file = unpack_dir.path().join("version");
+    let mut snapshot_version = String::new();
+    File::open(unpacked_version_file).and_then(|mut f| f.read_to_string(&mut snapshot_version))?;
+
+    Ok((
+        unpack_dir,
+        unpacked_snapshots_dir,
+        unpacked_append_vec_map,
+        snapshot_version,
+        measure_untar,
+    ))
+}
+
+/// Check if an incremental snapshot is compatible with a full snapshot.  This function parses the
+/// paths to see if the incremental snapshot's base slot is the same as the full snapshot's slot.
+/// Return an error if they are incompatible (or if the paths cannot be parsed), otherwise return a
+/// tuple of the full snapshot slot and the incremental snapshot slot.
+fn check_are_snapshots_compatible<P, Q>(
+    full_snapshot_archive_path: P,
+    incremental_snapshot_archive_path: Q,
+) -> Result<(Slot, Slot)>
```

Comment:
Since it seems the tuple from this check is never actually used, we could either:

1) just return a bool and have the caller return the `SnapshotError::IncompatibleSnapshots`

2) Return a `Result<()>`

Reply:
Ah good catch! This was leftover when I used the return values to set the last full snapshot slot and last incremental snapshot slot. I'll remove those here.

I think I may need them again when doing the `clean_accounts()` work, but that'll be easy to re-add then.

Reply:
Done!